test/squid_test.rb

```diff
@@ -0,0 +1,785 @@
+#! /usr/bin/env rspec
+
+require_relative "./test_helper"
+
+Yast.import "Squid"
+
+describe "Yast::Squid" do
+  subject(:squid) { Yast::Squid }
+
+  describe "#Write" do
+    subject(:squid) { Yast::SquidClass.new }
+
+    before do
+      allow(Yast::Progress).to receive(:New)
+      allow(Yast::Progress).to receive(:NextStage)
+      allow(Yast::Report).to receive(:Error)
+
+      allow(Yast2::SystemService).to receive(:find).with("squid").and_return(service)
+
+      allow(Yast::Mode).to receive(:auto) { auto }
+      allow(Yast::Mode).to receive(:commandline) { commandline }
+
+      allow(squid).to receive(:Abort).and_return(false)
+      allow(squid).to receive(:writeAllSettings).and_return(settings_written)
+      allow(squid).to receive(:writeFirewallSettings).and_return(firewall_settings_written)
+
+      squid.main
+    end
+
+    let(:service) { instance_double(Yast2::SystemService, save: true) }
+
+    let(:auto) { false }
+    let(:commandline) { false }
+    let(:settings_written) { true }
+    let(:firewall_settings_written) { true }
+
+    shared_examples "old behavior" do
+      before do
+        squid.write_only = write_only
+        allow(squid).to receive(:IsServiceEnabled).and_return(service_enabled_on_startup)
+      end
+
+      let(:write_only) { true }
+      let(:service_enabled_on_startup) { true }
+
+      it "does not save the system service" do
+        expect(service).to_not receive(:save)
+
+        squid.Write
+      end
+
+      context "and service must be enable" do
+        it "calls to #EnableService" do
+          expect(squid).to receive(:EnableService)
+
+          squid.Write
+        end
+      end
+
+      context "and service must be disable" do
+        let(:service_enabled_on_startup) { false }
+
+        it "calls to #EnableService" do
+          expect(squid).to receive(:DisableService)
+
+          squid.Write
+        end
+      end
+
+      context "and only is writing settings" do
```

Comment:
I expect here to have let with write_only set to true, even if global one is above. It is just more explicit what is happening in tests and allows to change global mock value without breaking tests.

Reply:
Same here, I just moved these tests. I didn't even read them. :-)